compilation of the mc6809 module fails

from a fresh radare2 install in a docker (via a docker pull):
```
r2@radare:~$ r2pm -i mc6809
Already up-to-date.
Install Done For mc6809
checking build system type... x86_64-unknown-linux-gnu
checking host system type... x86_64-unknown-linux-gnu
checking target system type... x86_64-unknown-linux-gnu
checking for working directories... current
using prefix '/home/r2/.local/share/radare2/prefix'
checking for c compiler... gcc
Using PKGCONFIG: pkg-config
checking pkg-config flags for r_core... yes
checking for libewf ... no
checking pkg-config flags for libewf... no
checking pkg-config flags for unicorn... no
checking for libyara ... no
checking pkg-config flags for yara... no
creating ./Makefile
creating ./options.mk
cleaning temporally files... done

Final report:
 - PREFIX = /home/r2/.local/share/radare2/prefix
rm -f *.so *.o asm_x86_udis.o asm_x86_olly.o ../arch/x86/ollyasm/disasm.o ../arch/x86/ollyasm/asmserv.o ../arch/x86/ollyasm/assembl.o asm_x86_tab.o asm_armthumb.o ../arch/arm/armthumb.o ../arch/ppc/ppc_disasm/ppc_disasm.o asm_ppc_disasm.o asm_psosvm.o ../arch/psosvm/vmas/vmas.o asm_m68k_net.o ../arch/m68k/m68k_disasm/m68k_disasm.o asm_blackfin.o asm_x86_bea.o ../arch/x86/bea/BeaEngine.o asm_msil.o asm_nios.o ../arch/nios/gnu/safe-ctype.o ../arch/nios/gnu/cgen-bitset.o ../arch/nios/gnu/cgen-dis.o ../arch/nios/gnu/cgen-asm.o ../arch/nios/gnu/cgen-opc.o ../arch/nios/gnu/cpu-nios.o ../arch/nios/gnu/nios-dis.o ../arch/nios/gnu/nios-asm.o ../arch/nios/gnu/nios-desc.o ../arch/nios/gnu/nios-ibld.o ../arch/nios/gnu/nios-opc.o asm_swf.o ../arch/swf/swfdis.o asm_vc4.o ../arch/vc4/vc4-dis.o ../arch/vc4/vc4-asm.o ../arch/vc4/vc4-desc.o ../arch/vc4/vc4-ibld.o ../arch/vc4/vc4-opc.o ../arch/vc4/cgen-bitset.o ../arch/vc4/cgen-dis.o ../arch/vc4/cgen-asm.o ../arch/vc4/cgen-opc.o ../arch/vc4/libvc4/eval.o ../arch/vc4/libvc4/decode.o ../arch/vc4/libvc4/vc4_decode.o ../arch/vc4/libvc4/vc4_arch.o ../arch/vc4/libvc4/vc4_util.o asm_mc6809.o ../arch/z80_nc/z80.o ../arch/z80_nc/disasm.o ../arch/z80_nc/test.o asm_z80_nc.o asm_pyc.o ../arch/pyc/pyc_disasm.o asm_x86_zyan.o ../arch/x86/zyan/src/Decoder.o ../arch/x86/zyan/src/Formatter.o ../arch/x86/zyan/src/InstructionTable.o ../arch/x86/zyan/src/Mnemonic.o ../arch/x86/zyan/src/Register.o ../arch/x86/zyan/src/Utils.o ../arch/x86/zyan/src/Zydis.o asm_evm.o ../arch/evm/evm.o asm_atombios.o ../arch/atombios/atombios.o
Makefile	asm_mc6809.c	  asm_vc4.c	  asm_z80_nc.c	msil.mk        x86_bea.mk
armthumb.mk	asm_msil.c	  asm_x86_bea.c   atombios.mk	nios.mk        x86_olly.mk
asm_armthumb.c	asm_nios.c	  asm_x86_olly.c  blackfin.mk	ppc_disasm.mk  x86_tab.mk
asm_atombios.c	asm_ppc_disasm.c  asm_x86_tab.c   configure	psosvm.mk      x86_udis.mk
asm_blackfin.c	asm_psosvm.c	  asm_x86_udis.c  evm.mk	pyc.mk	       x86_zyan.mk
asm_evm.c	asm_pyc.c	  asm_x86_zyan.c  m68k_net.mk	swf.mk	       z80.mk
asm_m68k_net.c	asm_swf.c	  asm_z80.c	  mc6809.mk	vc4.mk	       z80_nc.mk
gcc -I/home/r2/.local/share/radare2/prefix/include -I/usr/include/libr -g -I/shlr -I../../include -I../arch/ -I../arch/include -Wall -shared -fPIC  -DLIL_ENDIAN=1 -I../arch/x86 -I../arch/x86/bea/ -I../arch/x86/bea/src -I../arch/x86/bea/include -DUint64=uint64_t -I../arch/swf -Iarch/vc4/libvc4/ -Iarch/include/ -I../arch/pyc -I../arch/x86/zyan/include -I../arch/evm   -c -o asm_mc6809.o asm_mc6809.c
asm_mc6809.c: In function 'mc6809_disassemble':
asm_mc6809.c:1069:11: error: incompatible type for argument 1 of 'strcpy'
   strcpy (op->buf_asm, "INVALID");
           ^~
In file included from /usr/include/libr/r_types.h:168:0,
                 from asm_mc6809.c:4:
/usr/include/string.h:125:14: note: expected 'char * restrict' but argument is of type 'RStrBuf {aka struct <anonymous>}'
 extern char *strcpy (char *__restrict __dest, const char *__restrict __src)
              ^~~~~~
asm_mc6809.c:1073:11: error: incompatible type for argument 1 of 'strcpy'
   strcpy (op->buf_asm, mc6809_opcode->name);
           ^~
In file included from /usr/include/libr/r_types.h:168:0,
                 from asm_mc6809.c:4:
/usr/include/string.h:125:14: note: expected 'char * restrict' but argument is of type 'RStrBuf {aka struct <anonymous>}'
 extern char *strcpy (char *__restrict __dest, const char *__restrict __src)
              ^~~~~~
asm_mc6809.c:1077:12: error: incompatible type for argument 1 of 'sprintf'
   sprintf (op->buf_asm,
            ^~
In file included from asm_mc6809.c:3:0:
/usr/include/stdio.h:366:12: note: expected 'char * restrict' but argument is of type 'RStrBuf {aka struct <anonymous>}'
 extern int sprintf (char *__restrict __s,
            ^~~~~~~
asm_mc6809.c:1082:12: error: incompatible type for argument 1 of 'sprintf'
   sprintf (op->buf_asm,
            ^~
In file included from asm_mc6809.c:3:0:
/usr/include/stdio.h:366:12: note: expected 'char * restrict' but argument is of type 'RStrBuf {aka struct <anonymous>}'
 extern int sprintf (char *__restrict __s,
            ^~~~~~~
asm_mc6809.c:1088:12: error: incompatible type for argument 1 of 'sprintf'
   sprintf (op->buf_asm,
            ^~
In file included from asm_mc6809.c:3:0:
/usr/include/stdio.h:366:12: note: expected 'char * restrict' but argument is of type 'RStrBuf {aka struct <anonymous>}'
 extern int sprintf (char *__restrict __s,
            ^~~~~~~
asm_mc6809.c:1093:12: error: incompatible type for argument 1 of 'sprintf'
   sprintf (op->buf_asm, "%s $%04x",
            ^~
In file included from asm_mc6809.c:3:0:
/usr/include/stdio.h:366:12: note: expected 'char * restrict' but argument is of type 'RStrBuf {aka struct <anonymous>}'
 extern int sprintf (char *__restrict __s,
            ^~~~~~~
asm_mc6809.c:1099:12: error: incompatible type for argument 1 of 'sprintf'
   sprintf (op->buf_asm, "%s $%04x", mc6809_opcode->name,
            ^~
In file included from asm_mc6809.c:3:0:
/usr/include/stdio.h:366:12: note: expected 'char * restrict' but argument is of type 'RStrBuf {aka struct <anonymous>}'
 extern int sprintf (char *__restrict __s,
            ^~~~~~~
asm_mc6809.c:1108:12: error: incompatible type for argument 1 of 'strcpy'
    strcpy (op->buf_asm, "INVALID");
            ^~
In file included from /usr/include/libr/r_types.h:168:0,
                 from asm_mc6809.c:4:
/usr/include/string.h:125:14: note: expected 'char * restrict' but argument is of type 'RStrBuf {aka struct <anonymous>}'
 extern char *strcpy (char *__restrict __dest, const char *__restrict __src)
              ^~~~~~
asm_mc6809.c:1116:13: error: incompatible type for argument 1 of 'strcpy'
     strcpy (op->buf_asm, "INVALID");
             ^~
In file included from /usr/include/libr/r_types.h:168:0,
                 from asm_mc6809.c:4:
/usr/include/string.h:125:14: note: expected 'char * restrict' but argument is of type 'RStrBuf {aka struct <anonymous>}'
 extern char *strcpy (char *__restrict __dest, const char *__restrict __src)
              ^~~~~~
asm_mc6809.c:1119:14: error: incompatible type for argument 1 of 'sprintf'
     sprintf (op->buf_asm,
              ^~
In file included from asm_mc6809.c:3:0:
/usr/include/stdio.h:366:12: note: expected 'char * restrict' but argument is of type 'RStrBuf {aka struct <anonymous>}'
 extern int sprintf (char *__restrict __s,
            ^~~~~~~
asm_mc6809.c:1130:11: error: incompatible type for argument 1 of 'strcpy'
   strcpy (op->buf_asm, mc6809_opcode->name);
           ^~
In file included from /usr/include/libr/r_types.h:168:0,
                 from asm_mc6809.c:4:
/usr/include/string.h:125:14: note: expected 'char * restrict' but argument is of type 'RStrBuf {aka struct <anonymous>}'
 extern char *strcpy (char *__restrict __dest, const char *__restrict __src)
              ^~~~~~
asm_mc6809.c:1131:43: error: incompatible type for argument 1 of 'mc6809_append_indexed_args'
   op->size += mc6809_append_indexed_args (op->buf_asm,
                                           ^~
asm_mc6809.c:830:12: note: expected 'char *' but argument is of type 'RStrBuf {aka struct <anonymous>}'
 static int mc6809_append_indexed_args (char *buf_asm, const ut8 *buf)
            ^~~~~~~~~~~~~~~~~~~~~~~~~~
asm_mc6809.c:1136:11: error: incompatible type for argument 1 of 'strcpy'
   strcpy (op->buf_asm, mc6809_opcode->name);
           ^~
In file included from /usr/include/libr/r_types.h:168:0,
                 from asm_mc6809.c:4:
/usr/include/string.h:125:14: note: expected 'char * restrict' but argument is of type 'RStrBuf {aka struct <anonymous>}'
 extern char *strcpy (char *__restrict __dest, const char *__restrict __src)
              ^~~~~~
asm_mc6809.c:1138:8: error: incompatible type for argument 2 of 'mc6809_append_pushpull_args'
        op->buf_asm,
        ^~
asm_mc6809.c:999:12: note: expected 'char *' but argument is of type 'RStrBuf {aka struct <anonymous>}'
 static int mc6809_append_pushpull_args(enum instruction_mode mode,
            ^~~~~~~~~~~~~~~~~~~~~~~~~~~
asm_mc6809.c:1142:12: error: incompatible type for argument 1 of 'sprintf'
   sprintf (op->buf_asm,
            ^~
In file included from asm_mc6809.c:3:0:
/usr/include/stdio.h:366:12: note: expected 'char * restrict' but argument is of type 'RStrBuf {aka struct <anonymous>}'
 extern int sprintf (char *__restrict __s,
            ^~~~~~~
<builtin>: recipe for target 'asm_mc6809.o' failed
make: *** [asm_mc6809.o] Error 1

```

--- a/.travis.yml
+++ b/.travis.yml
@@ -43,6 +43,7 @@ env:
   - TESTS="r2pm -i r2frida"
 #  - TESTS="r2pm -i io-ewf" # https://github.com/travis-ci/apt-package-whitelist/issues/3205
 #  - TESTS="r2pm -i keystone-lib && r2pm -i keystone && cd radare2-regressions && make keystone && cd .." [BR] https://travis-ci.org/radare/radare2-extras/builds/157522362
+  - TESTS="r2pm -i mc6809"
   - TESTS="r2pm -i microblaze"
   - TESTS="r2pm -i msil"
   - TESTS="r2pm -i ppcdisasm"

--- a/libr/asm/p/asm_mc6809.c
+++ b/libr/asm/p/asm_mc6809.c
@@ -827,8 +827,7 @@ static const char mc6809_index_registers[] = {
 	/* 0b11 */ 's',
 };
 
-static int mc6809_append_indexed_args (char *buf_asm, const ut8 *buf)
-{
+static int mc6809_append_indexed_args(RStrBuf *buf_asm, const ut8 *buf) {
 	char postop_buffer[32];
 	int postop_bytes = 0;
 
@@ -992,42 +991,43 @@ static int mc6809_append_indexed_args (char *buf_asm, const ut8 *buf)
 			}
 		}
 	}
-	strcat (buf_asm, postop_buffer);
+	r_strbuf_append (buf_asm, postop_buffer);
 	return postop_bytes;
 }
 
 static int mc6809_append_pushpull_args(enum instruction_mode mode,
-				       char *buf_asm,
+				       RStrBuf *buf_asm,
 				       const ut8 *opcode_args)
 {
-	strcat (buf_asm, " ");
+	r_strbuf_append (buf_asm, " ");
 
 	if (*opcode_args & 0x80) {
-		strcat (buf_asm, "pc,");
+		r_strbuf_append (buf_asm, "pc,");
 	}
 	if (*opcode_args & 0x40) {
-		strcat (buf_asm, (mode == PUSHPULLSYSTEM) ? "u," : "s,");
+		r_strbuf_append (buf_asm, (mode == PUSHPULLSYSTEM) ? "u," : "s,");
 	}
 	if (*opcode_args & 0x20) {
-		strcat (buf_asm, "y,");
+		r_strbuf_append (buf_asm, "y,");
 	}
 	if (*opcode_args & 0x10) {
-		strcat (buf_asm, "x,");
+		r_strbuf_append (buf_asm, "x,");
 	}
 	if (*opcode_args & 0x08) {
-		strcat (buf_asm, "dp,");
+		r_strbuf_append (buf_asm, "dp,");
 	}
 	if (*opcode_args & 0x04) {
-		strcat (buf_asm, "b,");
+		r_strbuf_append (buf_asm, "b,");
 	}
 	if (*opcode_args & 0x02) {
-		strcat (buf_asm, "a,");
+		r_strbuf_append (buf_asm, "a,");
 	}
 	if (*opcode_args & 0x01) {
-		strcat (buf_asm, "cc,");
+		r_strbuf_append (buf_asm, "cc,");
 	}
 	/* Trim off the final unwanted comma */
-	buf_asm[strlen(buf_asm)-1] = '\0';
+	/* XXX we miss a proper r_str_buf_trim function here */
+	r_strbuf_get (buf_asm)[r_strbuf_length(buf_asm)-1] = '\0';
 	return 2;
 }
 
@@ -1035,7 +1035,7 @@ static int mc6809_disassemble(RAsm *a, RAsmOp *op, const ut8 *buf, int len) {
 	ut8 tfrexg_regmasked;
 	const char *tfrexg_source_reg;
 	const char *tfrexg_dest_reg;
-
+	const char *buf_asm = "invalid";
 
 	const mc6809_opcodes_t *mc6809_opcode = &mc6809_opcodes[buf[0]];
 	/* opcode_args points to the first argument byte of the opcode */
@@ -1066,37 +1066,33 @@ static int mc6809_disassemble(RAsm *a, RAsmOp *op, const ut8 *buf, int len) {
 	case PAGE2: /* PAGE2 and PAGE3 shouldn't occur twice in a row */
 	case PAGE3:
 		op->size++;
-		strcpy (op->buf_asm, "INVALID");
 		break;
 	case INHERENT:
 		op->size++;
-		strcpy (op->buf_asm, mc6809_opcode->name);
+		buf_asm = sdb_fmt ("%s", mc6809_opcode->name);
 		break;
 	case IMMEDIATE:
 		op->size += 2;
-		sprintf (op->buf_asm,
-			 "%s #$%02x", mc6809_opcode->name, *opcode_args);
+		buf_asm = sdb_fmt ("%s #$%02x", mc6809_opcode->name, *opcode_args);
 		break;
 	case IMMEDIATELONG:
 		op->size += 3;
-		sprintf (op->buf_asm,
-			"%s #$%04x", mc6809_opcode->name,
+		buf_asm = sdb_fmt ("%s #$%04x", mc6809_opcode->name,
 			opcode_args[0] * 256 + opcode_args[1]);
 		break;
 	case DIRECT:
 		op->size += 2;
-		sprintf (op->buf_asm,
-			 "%s <$%02x", mc6809_opcode->name, *opcode_args);
+		buf_asm = sdb_fmt ("%s <$%02x", mc6809_opcode->name, *opcode_args);
 		break;
 	case RELATIVE:
 		op->size += 2;
-		sprintf (op->buf_asm, "%s $%04x",
+		buf_asm = sdb_fmt ("%s $%04x",
 			mc6809_opcode->name,
 			(ut16) (a->pc + (st8) *opcode_args + op->size) & 0xFFFF);
 		break;
 	case RELATIVELONG:
 		op->size += 3;
-		sprintf (op->buf_asm, "%s $%04x", mc6809_opcode->name,
+		buf_asm = sdb_fmt ("%s $%04x", mc6809_opcode->name,
 			(ut16) (a->pc + (st16)(opcode_args[0]*256+opcode_args[1])+op->size) & 0xFFFF);
 		break;
 	case TFREXG:
@@ -1105,7 +1101,6 @@ static int mc6809_disassemble(RAsm *a, RAsmOp *op, const ut8 *buf, int len) {
 		tfrexg_regmasked = *opcode_args & 0x88;
 		if (tfrexg_regmasked && tfrexg_regmasked != 0x88) {
 			op->size += 1;
-			strcpy (op->buf_asm, "INVALID");
 		} else {
 			tfrexg_source_reg = \
 				mc6809_register_field[(*opcode_args >> 4) & 0x0f];
@@ -1113,10 +1108,9 @@ static int mc6809_disassemble(RAsm *a, RAsmOp *op, const ut8 *buf, int len) {
 				mc6809_register_field[*opcode_args & 0x0f];
 			if (!tfrexg_source_reg || !tfrexg_dest_reg) {
 				op->size += 1;
-				strcpy (op->buf_asm, "INVALID");
 			} else {
 				op->size += 2;
-				sprintf (op->buf_asm,
+				buf_asm = sdb_fmt (
 					 "%s %s,%s",
 					 mc6809_opcode->name,
 					 tfrexg_source_reg,
@@ -1127,25 +1121,27 @@ static int mc6809_disassemble(RAsm *a, RAsmOp *op, const ut8 *buf, int len) {
 		break;
 	case INDEXED:
 		/* Load Effective Address opcode - variable length */
-		strcpy (op->buf_asm, mc6809_opcode->name);
-		op->size += mc6809_append_indexed_args (op->buf_asm,
+		buf_asm = sdb_fmt ("%s", mc6809_opcode->name);
+		op->size += mc6809_append_indexed_args (&op->buf_asm,
 							opcode_args) + 1;
 		break;
 	case PUSHPULLSYSTEM:
 	case PUSHPULLUSER:
-		strcpy (op->buf_asm, mc6809_opcode->name);
+		buf_asm = sdb_fmt ("%s", mc6809_opcode->name);
 		op->size += mc6809_append_pushpull_args(mc6809_opcode->mode,
-							op->buf_asm,
+							&op->buf_asm,
 							opcode_args);
 		break;
 	case EXTENDED:
-		sprintf (op->buf_asm,
+		buf_asm = sdb_fmt (
 			 "%s $%04x",
 			 mc6809_opcode->name,
 			 opcode_args[0] * 256 + opcode_args[1]);
 		op->size += 3;
 		break;
 	}
+
+	r_asm_op_set_asm (op, buf_asm);
 
 	return op->size;
 }

